client/hotrod/src/test/java/org/infinispan/hotrod/HotRodMutinyCacheTest.java

```diff
@@ -0,0 +1,280 @@
+package org.infinispan.hotrod;
+
+import static org.infinispan.hotrod.AwaitAssertions.assertAwaitEquals;
+import static org.infinispan.hotrod.AwaitAssertions.await;
+import static org.infinispan.hotrod.CacheEntryAssertions.assertEntry;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.smallrye.mutiny.Multi;
+import org.infinispan.api.common.CacheEntry;
+import org.infinispan.api.common.CacheEntryMetadata;
+import org.infinispan.api.common.CacheEntryVersion;
+import org.infinispan.api.common.CacheWriteOptions;
+import org.infinispan.hotrod.impl.cache.CacheEntryImpl;
+import org.infinispan.hotrod.impl.cache.CacheEntryMetadataImpl;
+import org.infinispan.hotrod.impl.cache.CacheEntryVersionImpl;
+import org.infinispan.hotrod.test.AbstractMutinyCacheSingleServerTest;
+import org.infinispan.hotrod.test.KeyValueGenerator;
+import org.infinispan.hotrod.util.FlowUtils;
+import org.infinispan.hotrod.util.MapKVHelper;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * @since 14.0
+ **/
+public class HotRodMutinyCacheTest<K, V> extends AbstractMutinyCacheSingleServerTest<K, V> {
+
+   @MethodSource("parameterized")
+   @ParameterizedTest(name = "getPut[{0}]")
+   public void testPut(KeyValueGenerator<K, V> kvGenerator) {
+      final K key = kvGenerator.generateKey(cacheName, 0);
+      final V v1 = kvGenerator.generateValue(cacheName, 0);
+
+      final CacheWriteOptions options = CacheWriteOptions.writeOptions()
+            .timeout(Duration.ofSeconds(15))
+            .lifespanAndMaxIdle(Duration.ofSeconds(20), Duration.ofSeconds(25))
+            .build();
+      final CacheEntryVersion version1 = new CacheEntryVersionImpl(1);
+      assertEntry(key, null, kvGenerator, await(cache.put(key, v1, options)));
+      kvGenerator.assertValueEquals(v1, await(cache.get(key)));
```

Comment:
would not be better to use Mutiny await instead of await?

Reply:
Nice, going to use it and unwrap the Uni within the static `await`.